subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/asm/ClassDependenciesVisitor.java

```diff
@@ -218,26 +235,27 @@ protected MethodVisitor(Set<String> types) {
 
         @Override
         public void visitLocalVariable(String name, String desc, String signature, Label start, Label end, int index) {
-            maybeAddDependentType(types, Type.getType(desc));
+            maybeAddClassTypesFromSignature(signature, privateTypes);
+            maybeAddDependentType(privateTypes, Type.getType(desc));
             super.visitLocalVariable(name, desc, signature, start, end, index);
         }
 
         @Override
         public org.objectweb.asm.AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
```

Comment:
Sorry for trouble with that but can we revert these? `visitLocalVariable ` is ok, but these `visitAnnotation`, `visitParameterAnnotation` and  `visitParameterAnnotation ` might not be.

I checked on some demo Micronaut project and annotation processor checks also annotations for parameters from super type. So I believe this change would not be ok for annotation processor that needs some more info from super class.

Reply:
Done.